### PR TITLE
Make openstack role OASIS-compatible

### PR DESCRIPTION
**Pull request by 14rcole**
I converted this role while perfecting the [automation](github.com/14rcole/oasis-scripts) for converting the roles.  After running the scripts, you still have to run `molecule lint -s docker` in the role directory and fix any errors manually

### CODE REVIEW

File: linchpin/provision/roles/openstack/tasks/provision_os_keypair.yml [hunk 1]
```diff
@@ -10,21 +10,30 @@
...
+    msg: "{{ res_def_output_nasync['key']['private_key'] |
+             write_to_file(default_ssh_key_path+'/', res_def['name']+'.key') }}"
+  when: state == "present" and
+        res_def_output_nasync['changed'] and
+        not _async
 
 # commented due to selinux issues
 #- name: "Generate keyfile"
-#  copy: 
+#  copy:
```
Comment (samvarankashyap) [inline linchpin/provision/roles/openstack/tasks/provision_os_keypair.yml]: @14rcole You may delete these commented lines

File: config/Dockerfiles/Jenkinsfile [hunk 2]
```diff
@@ -4,8 +4,7 @@ env.ghprbGhRepository = env.ghprbGhRepository ?: 'CentOS-PaaS-SIG/linchpin'
 env.ghprbActualCommit = env.ghprbActualCommit ?: 'develop'
 env.ghprbSourceBranch = env.ghprbSourceBranch ?: 'develop'
 env.ghprbTargetBranch = env.ghprbTargetBranch ?: 'develop'
-env.ghprbPullAuthorLogin = env.ghprbPullAuthorLogin ?: ''
-env.ghprbPullId = env.ghprbPullId ?: ''
+env.ghprbPullAuthorLogin = env.ghprbPullAuthorLogin ?: '' env.ghprbPullId = env.ghprbPullId ?: ''
```
Comment (p3ck) [inline config/Dockerfiles/Jenkinsfile]: this shouldn't all be on one line should it?

File: linchpin/provision/roles/openstack/tasks/provision_os_sg.yml [hunk 3]
```diff
@@ -10,7 +9,7 @@
 
 - name: "provision/teardown security group"
   os_security_group_rule:
-    auth: "{{ group.0 }}"
+    # auth: "{{ group.0 }}"
```
Comment (samvarankashyap) [inline linchpin/provision/roles/openstack/tasks/provision_os_sg.yml]: @14rcole You may delete this line too

Comment (14rcole) [reply linchpin/provision/roles/openstack/tasks/provision_os_sg.yml]: Okay, I wanted to make sure I was commenting the right one

File: config/Dockerfiles/Jenkinsfile (hunk 2 above)
Comment (14rcole) [reply config/Dockerfiles/Jenkinsfile]: That shouldn't have been changed at all, thanks for the catch

**Comment by samvarankashyap** [comment]
@14rcole The tests hung up on Jenkins. I had to cancel the job. 
an interesting observation is the failure msg shows it failed on centos8 
though we have not added support for centos8.

**Comment by samvarankashyap** [comment]
[test]

**Comment by 14rcole** [comment]
That's strange.  The CentOS 8 PR is also failing for reasons I can't determine

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
@14rcole I don't see any issue on GitHub relating to this PR. 
could you make a reference to it. Thanks

**Comment by 14rcole** [comment]
Fixes #1367

**Comment by 14rcole** [comment]
@samvarankashyap @abraverm PTAL

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**